plugins/advisors/black-duck/src/funTest/kotlin/ResponseCachingComponentServiceClient.kt

```diff
@@ -37,15 +38,19 @@ import java.util.concurrent.ConcurrentHashMap
  * as a fake [ComponentServiceClient].
  */
 internal class ResponseCachingComponentServiceClient(
-    private val overrideFile: File,
+    overrideUrl: URL?,
     private val serverUrl: String?,
     apiToken: String?
 ) : ComponentServiceClient {
     // The Black Duck library uses GSON to serialize its POJOs. So use GSON, too, because this is the simplest option.
     private val gson = GsonBuilder().setPrettyPrinting().create()
 
-    private val cache = if (overrideFile.isFile) {
-        gson.fromJson(overrideFile.readText(), ResponseCache::class.java)
+    private val overrideFile = overrideUrl?.takeIf { it.protocol == "file" }?.let { File(it.path) }
```

Comment:
Maybe we can factor out the utility `fun URL.asFile(): File? = takeIf { it.protocol == "file" }?.let { File(it.path) }`

Reply:
Hm, isn't it clear already that this will be re-used, and also already now a tad more readable? Anyhow, not insisting on it.